### DISCUSSION TITLE
Using zoom and pan events to trigger zoom and pan of another component

### DISCUSSION THREAD

**Discussion by sabres207** [chartjs/chartjs-plugin-zoom#542] [Q&A]
Hi all,
I'm trying to achieve a functionality of some other component we use inside our company to be able to get zoomed and panned upon events fired by chart js zoom plugin.

For example the event:
onZoom: chart => { // do something }

I tried to figure out how the zoom state of the chart is represented in the chart argument that is sent to the onZoom callback, but I couldnt find anything special representing where on the chart am I in right now.
What I expected to see is something like the points of the dataset that are visible at the moment, or some kind of coordinates or the exact percentage of the start and end point that is visible at the moment.

Would appreciate any help. Thanks!

**Comment by kurkle** [comment]
You can check the scale min/max values to see what range is visible. Would that help?

**Comment by ronildo** [comment]
> Hi all, I'm trying to achieve a functionality of some other component we use inside our company to be able to get zoomed and panned upon events fired by chart js zoom plugin.
> 
> For example the event: onZoom: chart => { // do something }
> 
> I tried to figure out how the zoom state of the chart is represented in the chart argument that is sent to the onZoom callback, but I couldnt find anything special representing where on the chart am I in right now. What I expected to see is something like the points of the dataset that are visible at the moment, or some kind of coordinates or the exact percentage of the start and end point that is visible at the moment.
> 
> Would appreciate any help. Thanks!

Did you ever figure this out?
I'm struggling to find a solution to this problem.


> You can check the scale min/max values to see what range is visible. Would that help?
By the way, this doesn't work well...
For example, as I zoom in, I want to know what is the y min/max visible so I can adjust the chart so it doesn't get a lot of white space.

Is that even possible?
